app/controllers/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ProductsController < ApplicationController
+
+  def show
+    @products = Product.where(vendor_id: params[:id]).order(product_name: :asc)
+    render :index
+  end
+
+  def new 
+    @product = Product.new
+    @vendor = Vendor.find(params[:id])
+    render :new
+  end
+
+  def create
+    @product = Product.new(new_product_create_params[:product])
+    if @product.save
+      @product[:product_id] = @product[:id]
+      @product.save
+      redirect_to product_path(@product.vendor_id)
```

Comment:
Nice job checking to make sure the product saves! It's important to remember the things that can go wrong when defining your controller actions.